### ISSUE TITLE
new incap client API

### ISSUE THREAD

**Issue by gstevey** [gradle/incap#30]
### Milestone 1

incap splits into two libraries:  incap build (for build-side API) and incap for APs.

gradle modules:

  * incap/build
  * incap/ap

We think there are 2 primary API entry pointsfor incap-B(uild):

```java
  // given a file path, return the -A javac option for communicating that path
  //  - gradle will pass this option to incap-ap for where to write the file
  //  - this shows up in ProcessingEnvironment.init(), and the AP must pass it to incap-ap
  String getProcessorOptionForMappingFile(File mappingFile)
```
Note: we would prefer to receive a folder (`mappingFile` --> `mappingFolder`) here because we need to enforce that the name of the file contains the version number and we could do it here. @gstevey @oehme

```java
  // API for Gradle to read what incap-ap wrote after each build:
  // map of AP FQN to the AP's new mappings (FQN -> target description)
  // Note: incap ap has to do the merge of the separate sections (from separate rounds). 
  Map<String<Map<String,String>> readPairs(File filePath)
```
Please note: this file is finally not versioned because it is transient, it will be recreated after each full build.

--- 
### Milestone 2
Provide an API for writing out the full set of merged mappings;
e.g. something like:
```java
void writePairs(File outFile, Map<String, Map<String,String>, String version>
  throws IOException
```
If the latter exists, then the readPairs API is called twice per build:

  - once to read the full mappings file (for Gradle) (before the build starts)
  - once later to read the incremental mappings file (after processing ends)

**Comment by oehme** [comment]
The full mapping is not read by Gradle, it is read on the processor side. Gradle writes it.

**Comment by stephanenicolas** [comment]
Solved in #63.
